storage/src/main/java/tech/pegasys/teku/storage/client/ChainHead.java

```diff
@@ -49,10 +48,10 @@ public UInt64 getForkChoiceEpoch() {
     return spec.computeEpochAtSlot(forkChoiceSlot);
   }
 
-  public UInt64 findCommonAncestor(final ChainHead other) {
+  public SlotAndBlockRoot findCommonAncestor(final ChainHead other) {
     if (getSlot().equals(UInt64.ZERO) || other.getSlot().equals(UInt64.ZERO)) {
       // One fork has no blocks so the only possible common ancestor is genesis.
-      return UInt64.ZERO;
+      return new SlotAndBlockRoot(UInt64.ZERO, getBlockRootAtSlot(UInt64.ZERO));
```

Comment:
I'm not sure it's safe to use `getBlockRootAtSlot` here because the canonical fork may be long enough that slot zero is no longer available in the block root history in state.  We should just use the block root from whichever `ChainHead` has slot 0 since that must be the genesis block root.